src/analysis/normed_space/star/complex.lean

```diff
@@ -0,0 +1,87 @@
+/-
+Copyright (c) 2022 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import analysis.normed_space.star.basic
+import algebra.star.self_adjoint
+import analysis.complex.basic
+
+/-!
+# Complex normed star modules and algebras
+
+Facts about star modules and star algebras over the complex numbers.
+
+## Main definitions
+
+* `star_ring.re`: the real part of an element of a star module, defined as `2⁻¹ • (x + star x)`
+* `star_ring.im`: the imaginary part of an element of a star module, defined as
+  `(-I * 2⁻¹) • (x - star x)`.
+
+-/
+
+variables {E : Type*}
+
+namespace star_module
+open_locale complex_conjugate
+open complex
+
+variables [add_comm_group E] [star_add_monoid E] [module ℂ E] [star_module ℂ E]
+
+/-- The real part of an element of a star module, as a real-linear map. -/
+@[simps] noncomputable def re : E →ₗ[ℝ] self_adjoint E :=
+{ to_fun := λ x, ⟨(2⁻¹ : ℂ) • (x + star x), by simp only [self_adjoint.mem_iff, star_smul, add_comm,
+                                        star_add_monoid.star_add, star_inv', star_bit0,
+                                        star_one, star_star]⟩,
+  map_add' := λ x y, by { refine subtype.ext_val _, simp, abel },
+  map_smul' := λ r x,
+    begin
+      refine subtype.ext_val _,
+      simp only [star_smul, is_R_or_C.star_def, is_R_or_C.conj_to_real, smul_add, ring_hom.id_apply,
+                 subtype.val_eq_coe, self_adjoint.coe_smul, add_subgroup.coe_mk],
+      rw [smul_comm],
+      refine congr_arg _ _,
+      rw [smul_comm]
+    end }
```

Comment:
I think it's worth golfing the proofs in this file a bit. For example:
```suggestion
    star_add_monoid.star_add, star_inv', star_bit0, star_one, star_star]⟩,
  map_add' := λ x y, by { ext, simp [add_add_add_comm], },
  map_smul' := λ r x,
  begin
    ext,
    simp only [star_smul, is_R_or_C.star_def, is_R_or_C.conj_to_real, smul_add,
      add_subgroup.coe_mk, ring_hom.id_apply, self_adjoint.coe_smul, smul_comm (2⁻¹ : ℂ) r],
  end }
```
and similarly below.

Reply:
Actually the `map_smul'` proof can probably even be just `ext, simp [smul_comm (2⁻¹ : ℂ) r]`, but I've just closed my gitpod session so I'll let you check 😉